components/profiles/profiles.go

```diff
@@ -0,0 +1,85 @@
+package profiles
+
+import (
+	dsc "github.com/opendatahub-io/opendatahub-operator/apis/datasciencecluster/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/components/dashboard"
+	"github.com/opendatahub-io/opendatahub-operator/components/datasciencepipelines"
+	"github.com/opendatahub-io/opendatahub-operator/components/modelmeshserving"
+	"github.com/opendatahub-io/opendatahub-operator/components/workbenches"
+)
+
+const (
+	ProfileFull      = "full"
+	ProfileServing   = "serving"
+	ProfileTraining  = "training"
+	ProfileWorkbench = "workbench"
+)
+
+type ReconciliationPlan struct {
+	Components map[string]bool
+}
+
+type ProfileConfig struct {
+	ComponentDefaults map[string]bool
+}
+
+var profileConfigs = map[string]ProfileConfig{
+	ProfileServing: {
+		ComponentDefaults: map[string]bool{
+			modelmeshserving.ComponentName:     true,
+			datasciencepipelines.ComponentName: false,
+			workbenches.ComponentName:          false,
+			dashboard.ComponentName:            true,
+		},
+	},
+	ProfileTraining: {
+		ComponentDefaults: map[string]bool{
+			modelmeshserving.ComponentName:     false,
+			datasciencepipelines.ComponentName: true,
+			workbenches.ComponentName:          false,
+			dashboard.ComponentName:            true,
+		},
+	},
+	ProfileWorkbench: {
+		ComponentDefaults: map[string]bool{
+			modelmeshserving.ComponentName:     false,
+			datasciencepipelines.ComponentName: false,
+			workbenches.ComponentName:          true,
+			dashboard.ComponentName:            true,
+		},
+	},
+	ProfileFull: {
+		ComponentDefaults: map[string]bool{
+			modelmeshserving.ComponentName:     true,
+			datasciencepipelines.ComponentName: true,
+			workbenches.ComponentName:          true,
+			dashboard.ComponentName:            true,
+		},
+	},
+	// Add more profiles and their component defaults as needed
+}
+
+func CreateReconciliationPlan(instance *dsc.DataScienceCluster) *ReconciliationPlan {
+	plan := &ReconciliationPlan{}
+
+	profile := instance.Spec.Profile
+	if profile == "" {
+		profile = ProfileFull
+	}
+
+	setServingProfile(profileConfigs, plan, instance)
+	// Similarly set other profiles
+	return plan
+}
+
+func setServingProfile(profiledefaults map[string]ProfileConfig, plan *ReconciliationPlan, instance *dsc.DataScienceCluster) {
+
+	// serving is enabled by default, unless explicitly overriden
```

Comment:
I think this logic does not work, as any component that is disabled in the profile will evaluate the expression to false. See my [previous commit](https://github.com/VaishnaviHire/opendatahub-operator/pull/6/commits/888d0669e40352cd05ff48fdb20e77f06f8efa39#diff-a28bb610b027c13a4be8b3dc6351d8e75e3d8b593daab115d3a4719c28535b22R74) for a suggestion for the code.